pages/journal/[id].tsx

```diff
@@ -1,159 +1,172 @@
+                    margin-bottom: 10px;
+                }
+
+                .datePublished {
+                    color: #999;
+                    font-size: 17px;
+                    margin-bottom: 30px;
+                }
+
+                .blogContent::first-letter {
+                    font-size: 150%;
+                    font-weight: bold;
+                }
+
+                .blogContent {
+                    font-size: 18px;
+                    margin: 50px 0;
+                }
+
+                .relatedContainer {
+                    width: 90%;
+                    margin-left: -45%;
+                    position: relative;
+                    left: 50%;
+                }
+
+                .relatedTitle {
+                    font-size: 34px;
+                    color: #707070;
+                }
+
+                .thumbnailGrid {
+                    width: 100%;
+                    display: grid;
+                    grid-template-columns: repeat(auto-fill, 400px);
+                    justify-content: center;
+                    grid-gap: 1rem;
+                    margin-bottom: 50px;
+                }
+
+                a {
+                    text-decoration: inherit;
+                    color: inherit;
+                }
+
+                @media only screen and (min-width: 770px) {
+                    .blogContainer {
+                        width: 60%;
+                        margin-left: 5%;
+                        left: 0;
+                    }
+                    .thumbnailGrid {
+                        justify-content: space-between;
+                    }
+                }
+
+                @media only screen and (min-width: 1200px) {
+                    .blogContainer {
+                        width: 50%;
+                    }
+                }
+            `}</style>
+        </main>
+    );
 };
 
-export async function getServerSideProps(context: NextPageContext) {
-  var post: JournalEntry = await getPostById(context.query.id)
-  var related: JournalEntry[] = await getEntriesByType(post.category)
-  // console.log(related)
-  return {
-    props: {
-      post: post,
-      relatedEntries: related.filter((entry) => entry.id !== post.id)
-    }
-  }
+export async function getStaticProps(context: GetStaticPropsContext) {
+    const post: JournalEntry = await getJournalEntryById(context.params?.id as string);
+    const related: JournalEntry[] = await getJournalEntryByType(post.category as string);
+    // console.log(related)
+    return {
+        props: {
+            post: post,
+            relatedEntries: related.filter(entry => entry.id !== post.id),
+        },
+        revalidate: 15,
+    };
+}
+
+export async function getStaticPaths() {
+    const posts: JournalEntry[] = await getJournalEntriesByReviewStatus(true);
+
+    const paths = posts.map(post => ({
+        params: { id: post.id },
+    }));
+
+    return { paths, fallback: false };
```

Comment:
Do we want fallback to be true here since new journals might be added after the build?